src/main/kotlin/com/jetbrains/snakecharm/inspections/SmkRuleSectionAfterExecutionInspection.kt

```diff
@@ -50,36 +48,36 @@ class SmkRuleSectionAfterExecutionInspection : SnakemakeInspection() {
                                 ),
                                 ProblemHighlightType.ERROR,
                                 null,
-                                SwapTwoSectionsQuickFix(SmartPointerManager.createPointer(executionSection))
+                                MoveExecutionSectionToEndOfRuleQuickFix(SmartPointerManager.createPointer(executionSection))
                         )
                     }
                 }
             }
         }
     }
 
-    private class SwapTwoSectionsQuickFix(private val precedingSectionPointer: SmartPsiElementPointer<SmkSection>) : LocalQuickFix {
-        override fun getFamilyName() = SnakemakeBundle.message("INSP.INTN.move.rule.section.upwards.family")
+    private class MoveExecutionSectionToEndOfRuleQuickFix(
+            private val executionSectionPointer: SmartPsiElementPointer<SmkSection>
+    ) : LocalQuickFix {
+        override fun getFamilyName() = SnakemakeBundle.message("INSP.INTN.move.execution.section.down.family")
 
         override fun applyFix(project: Project, descriptor: ProblemDescriptor) {
-            val firstSection = precedingSectionPointer.element ?: return
-            val secondSection = descriptor.psiElement
+            val executionSection = executionSectionPointer.element ?: return
```

Comment:
This `return` means that the user selects the quick fix, presses `apply quick fix` and nothing happens?

Reply:
I guess so. This is to avoid further nullability checks and/or NPE. Normally `element` should not return null, but if in some unforeseeable case it does (documentation: `@return the PSI element, or null if the PSI reparse has completely invalidated the pointer (for example, the element referenced by the pointer has been deleted).`; I am not sure this might happen here, but I am also not sure that it can't ever happen here), what would be a better way to go about it? I don't think displaying an exception to the user is good here, should there be a pop-up message with text "unable to apply quick fix" or something like that?

Reply:
Ok, seems quick fixes doesn't provide API like `isEnabled` / `isValid`.